File: methodoverriding.cpp
#include <iostream>

class Book {
public:
    void display() {
        std::cout << "Name-The complete Ref Java 2" << std::endl;
        std::cout << "Author-Herbert Schildt" << std::endl;
        std::cout << std::endl;
    }
};

class Reference_Book : public Book {
public:
    void display() {
        std::cout << "Name-Pure Java 2" << std::endl;
        std::cout << "Author-Kenneth L" << std::endl;
        std::cout << std::endl;
    }
};

class Magazine : public Book {
public:
    void display() {
        std::cout << "Name-Core Java" << std::endl;
        std::cout << "Author-Cay S" << std::endl;
        std::cout << std::endl;
    }
};

int main() {
    Book book;
    Reference_Book referenceBook;
    Magazine magazine;

    std::cout << "Displaying Book:" << std::endl;
    book.display();

    std::cout << "Displaying Reference_Book:" << std::endl;
    referenceBook.display();

    std::cout << "Displaying Magazine:" << std::endl;
    magazine.display();

    return 0;
}

File: methodoverriding.java
public class InheritanceDemo
{
public static void main(String[] args)
{
Book b;
b=new Book();
b.display();
b=new Reference_Book();
b.display();
b=new Magazine();
b.display();
}
}
class Book
{
void display() {
System.out.println("Name-The complete Ref Java 2"); 
System.out.println("Author-Herbert Schildt");
System.out.println();
}
}
class Reference_Book extends Book
{
void display(){
System.out.println("Name-Pure Java 2"); 
System.out.println("Author-Kenneth L");
System.out.println();
}
}
class Magazine extends Book
{
void display(){
System.out.println("Name-Core Java"); 
System.out.println("Author-Cay S");
System.out.println();
}
